source/h5vol/H5Vol_attr.c

```diff
@@ -154,29 +154,39 @@ void GetFromAttribute(void *attrObj, hid_t *ret_id, H5VL_attr_get_t get_type)
     }
 }
 
-herr_t H5VL_adios2_attr_get(void *obj, H5VL_attr_get_t get_type, hid_t dxpl_id,
-                            void **req, va_list arguments)
+herr_t H5VL_adios2_attr_get(void *obj, H5VL_attr_get_args_t *args, hid_t dxpl_id,
+                            void **req)
 {
     REQUIRE_NOT_NULL_ERR(obj, -1);
     H5VL_ObjDef_t *vol = (H5VL_ObjDef_t *)obj;
-
-    if ((get_type == H5VL_ATTR_GET_SPACE) || (get_type == H5VL_ATTR_GET_TYPE))
+ 
+    switch (args->op_type)
     {
-        hid_t *ret_id = va_arg(arguments, hid_t *);
-        GetFromAttribute((vol->m_ObjPtr), ret_id, get_type);
+    case H5VL_ATTR_GET_SPACE:
+      {
+        hid_t *ret_id = (hid_t *)args->args.get_space.space_id;
+        GetFromAttribute((vol->m_ObjPtr), ret_id, args->op_type);
         return 0;
+      }
+    case H5VL_ATTR_GET_TYPE:
+      {
+        hid_t *ret_id = (hid_t *)args->args.get_type.type_id;
+        GetFromAttribute((vol->m_ObjPtr), ret_id, args->op_type);
+        return 0;
+      }
+    default:
+      break;
     }
 
-    const H5VL_loc_params_t *loc_params =
-        va_arg(arguments, const H5VL_loc_params_t *);
+    const H5VL_loc_params_t *loc_params = &args->args.get_info.loc_params;
```

Comment:
I would probably move this inside the get_info block so that you don't try to access the get_info portion when that's not the operation being performed.